app/models/spree/user.rb

```diff
@@ -125,7 +125,12 @@ def can_own_more_enterprises?
     end
 
     def default_card
-      credit_cards.where(is_default: true).first
+      # Don't re-fetch associated cards from the DB if they're already eager-loaded
+      if credit_cards.loaded?
+        credit_cards.to_a.find(&:is_default)
+      else
+        credit_cards.where(is_default: true).first
```

Comment:
have you tried defining this as an association and using AR association's [:condition](https://apidock.com/rails/v3.2.13/ActiveRecord/Associations/ClassMethods/has_one)? This feels to me like something AR should do and not us. 

We can improve it in a different PR if you prefer. I don't want to delay this PR.